repo/common/common.go

```diff
@@ -21,7 +21,7 @@ func MapGetKV(v map[string]interface{}, key string) (interface{}, error) {
 
 		cursor, ok = mcursor[part]
 		if !ok {
-			return nil, fmt.Errorf("%s key has no attributes", sofar)
+			return nil, fmt.Errorf("%s.%s not found", sofar, part)
```

Comment:
When it fails to get a root key because it doesn't exist, I get back an odd path.
```
$ ipfs config --json MissingKey.Something
Error: failed to get config value: ".MissingKey not found"
```

I don't know of a small trivial one-liner to avoid writing the period if `sofar` is empty, but something like the following would print "MissingKey not found"
```golang
var path string
if len(sofar) > 0 {
    path += sofar + "."
}
path += part
return nil, fmt.Errorf("%s not found", path)
```
```
$ ipfs config --json MissingKey.Something
Error: failed to get config value: "MissingKey not found"
```

Reply:
An alternative, we can set `sofar` to some root notation if it's empty. Maybe something like `$` or `Root`.
```golang
if len(sofar) == 0 {
    sofar = "$" // Denote that the key that is missing is a root key
}
return nil, fmt.Errorf("%s.%s not found", sofar, part)
```
```
$ ipfs config --json MissingKey.Something
Error: failed to get config value:  "$.MissingKey not found"
```